src/mono/dbi/cordb-function.cpp

```diff
@@ -0,0 +1,210 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//
+// File: CORDB-FUNCTION.CPP
+//
+
+#include <cordb-assembly.h>
+#include <cordb-code.h>
+#include <cordb-function.h>
+#include <cordb-process.h>
+#include <cordb.h>
+
+using namespace std;
+
+CordbFunction::CordbFunction(Connection* conn, mdToken token, int id, CordbModule* module) : CordbBaseMono(conn)
+{
+    this->m_metadataToken = token;
+    this->m_debuggerId    = id;
+    m_pCode               = NULL;
+    this->m_pModule       = module;
+    if (module)
+        module->InternalAddRef();
+    conn->GetProcess()->AddFunction(this);
+}
+
+CordbFunction::~CordbFunction()
+{
+    if (m_pCode)
+        m_pCode->InternalRelease();
+    if (m_pModule)
+        m_pModule->InternalRelease();
+}
+
+HRESULT CordbFunction::QueryInterface(REFIID id, void** pInterface)
+{
+    if (id == IID_ICorDebugFunction)
+    {
+        *pInterface = static_cast<ICorDebugFunction*>(this);
+    }
+    else if (id == IID_ICorDebugFunction2)
+    {
+        *pInterface = static_cast<ICorDebugFunction2*>(this);
+    }
+    else if (id == IID_ICorDebugFunction3)
+    {
+        *pInterface = static_cast<ICorDebugFunction3*>(this);
+    }
+    else if (id == IID_ICorDebugFunction4)
+    {
+        *pInterface = static_cast<ICorDebugFunction4*>(this);
+    }
+    else if (id == IID_IUnknown)
+    {
+        *pInterface = static_cast<IUnknown*>(static_cast<ICorDebugFunction*>(this));
+    }
+    else
+    {
+        *pInterface = NULL;
+        return E_NOINTERFACE;
+    }
+    AddRef();
+    return S_OK;
+}
+
+HRESULT CordbFunction::GetModule(ICorDebugModule** ppModule)
+{
+    LOG((LF_CORDB, LL_INFO100000, "CordbFunction - GetModule - IMPLEMENTED\n"));
+    HRESULT hr = S_OK;
+    EX_TRY 
+    {    
+        MdbgProtBuffer localbuf;
+        if (!m_pModule)
+        {
+            m_dbgprot_buffer_init(&localbuf, 128);
+            m_dbgprot_buffer_add_id(&localbuf, m_debuggerId);
+            int cmdId = conn->SendEvent(MDBGPROT_CMD_SET_METHOD, MDBGPROT_CMD_METHOD_ASSEMBLY, &localbuf);
+            m_dbgprot_buffer_free(&localbuf);
+
+            ReceivedReplyPacket* received_reply_packet = conn->GetReplyWithError(cmdId);
+            CHECK_ERROR_RETURN_FALSE(received_reply_packet);
+            MdbgProtBuffer* pReply = received_reply_packet->Buffer();
+
+            int module_id = m_dbgprot_decode_id(pReply->p, &pReply->p, pReply->end);
+            m_pModule     = conn->GetProcess()->GetModule(module_id);
+            if (m_pModule)
+                m_pModule->InternalAddRef();
+        }
+
+        if (!m_pModule)
```

Comment:
I would add `{` to the first `if`, and fix the new line at the `else {`

Reply:
Done.